pkg/plugins/instrumentationutils/request_status.go

```diff
@@ -0,0 +1,104 @@
+package instrumentationutils
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+)
+
+type RequestStatus int
+
+const (
+	RequestStatusOK RequestStatus = iota
+	RequestStatusCancelled
+	RequestStatusError
+)
+
+func (status RequestStatus) String() string {
+	names := [...]string{"ok", "cancelled", "error"}
+	if status < RequestStatusOK || status > RequestStatusError {
+		return ""
+	}
+
+	return names[status]
+}
+
+func RequestStatusFromError(err error) RequestStatus {
+	status := RequestStatusOK
+	if err != nil {
+		status = RequestStatusError
+		if errors.Is(err, context.Canceled) || grpcstatus.Code(err) == grpccodes.Canceled {
```

Comment:
The addition of `grpcstatus.Code(err) == grpccodes.Canceled` fixes the status to be set to `cancelled` rather than `error`.